index.js

```diff
@@ -5,23 +5,52 @@ const morseCodeDictionary = require("./morse-code-dictionary.json");
  * @param {String[]}  - An array of strings.
  * @returns {string[]} An array of strings.
  */
-function sortByStringLength() {}
-
+function sortByStringLength(sentence) {
+  return sentence.sort((a, b) => a.length - b.length)
+}
 /**
  * Returns an array of the word in all scrolling positions.
  * @param {String} word - A string.
  * @returns {string[]} An array of strings
  * Example: "Hello"
  * [ 'elloH', 'lloHe', 'loHel', 'oHell', 'Hello' ]
  */
-function textScroller() {}
+function textScroller(sentence) {
+  if (sentence.length === 0) {
+    return [];
+  }
+  const reversedWord = [];
+  let sentenceArray = sentence.split("")
+  for (let i = 0; i < sentence.length - 1; i++){
+    let rotatedWord = sentenceArray.shift() 
+    sentenceArray.push(rotatedWord)
+    reversedWord.push(sentenceArray.join(""))
```

Comment:
Great implementation gents - this is a classic pattern for string mainupulation.

Also keep this pattern in mind.  This is very similar to what we do with a data structure called a 'stack'.  What we are doing is `shift`ing the top item off of our list / stack, then performing some operation and continuing through the stack.  This works because every `shift` call shortens the length of the array.